lib/AST/DiagnosticEngine.cpp

```diff
@@ -1011,9 +1084,15 @@ void DiagnosticEngine::emitDiagnostic(const Diagnostic &diagnostic) {
 
 const char *DiagnosticEngine::diagnosticStringFor(const DiagID id,
                                                   bool printDiagnosticName) {
+  // TODO: Print diagnostic names from `localization`.
   if (printDiagnosticName) {
     return debugDiagnosticStrings[(unsigned)id];
   }
+  if (localization.get()) {
+    std::string diagMessage =
+          localization.get()->getMessageOr(id,diagnosticStrings[(unsigned)id]);
```

Comment:
let's lookup `diagnosticStrings[(unsigned)id]` beforehand and either pass it as a default or return. Also formatting.